### ISSUE TITLE
[python] Segfault on empty NumPy arrays

### ISSUE THREAD

**Issue by stephane-caron** [Simple-Robotics/proxsuite#6]
The following Python example segfaults:

```python
import numpy as np
import proxsuite

H = np.array([[65.0, -22.0, -16.0], [-22.0, 14.0, 7.0], [-16.0, 7.0, 5.0]])
g = np.array([-13.0, 15.0, 7.0])
A = np.array([])
b = np.array([])
C = np.array([])
u = np.array([])
l = np.array([])

results = proxsuite.proxqp.dense.solve(H, g, A, b, C, u, l)
print("optimal x: {}".format(results.x))
```

However, it runs fine if we replace all ``np.array([])`` by ``[]``.

**Comment by jcarpent** [comment]
@Bambade could you handle the case for empty input quantities?

**Comment by jcarpent** [comment]
@stephane-caron passing none arguments should fix your problem

**Comment by Bambade** [comment]
Thanks for raising the issue @stephane-caron. 
If specifying the 0 shape it works fine as well with both dense and sparse backends (cf #9). I will fix it for making it work as well with directly empty arguments.
```
import numpy as np
import proxsuite 
H = np.array([[65.0, -22.0, -16.0], [-22.0, 14.0, 7.0], [-16.0, 7.0, 5.0]])
g = np.array([-13.0, 15.0, 7.0])
A = np.zeros((0,3))
b = np.zeros(0)
C = np.zeros((0,3))
u = np.zeros(0)
l = np.zeros(0) 
results_dense = proxsuite.proxqp.dense.solve(H, g, A, b, C, u, l)
print("optimal x: {}".format(results_dense.x))
results_sparse = proxsuite.proxqp.sparse.solve(H, g, A, b, C, u, l)
print("optimal x: {}".format(results_sparse.x))
```

**Comment by stephane-caron** [comment]
@jcarpent ``None`` was my first guess when discovering the API, but it raises a type error:

```
    results = proxsuite.proxqp.dense.solve(H, g, None, None, None, None, None)
TypeError: solve(): incompatible function arguments. The following argument types are supported:
```

Same outcome with ``results = proxsuite.proxqp.dense.solve(H, g)``.

**Comment by Bambade** [comment]
I have made a PR for adding the possibility to initialize proxqp (for dense and sparse backends) with None entries. 
I have added as well your examples as supplementary unit tests (used as well in the documentation for explaining which possibilities exist for initializing the model).

**Comment by jcarpent** [comment]
Fixed by #12
